### ISSUE TITLE
[rest] Get all connections with filters

### ISSUE THREAD

**Issue by blu3beri** [hyperledger/aries-framework-javascript-ext#121]
While working on a project I found some niceties from ACA-Py that we should copy over. For example, you can filter the connections when you make the call to the endpoint (only retrieve the state == 'complete' records). I am not too sure on the design, maybe something like: 

```json
{
  ...
  "filters": {
    "state": "complete",
    ...
  }
}
```

**Comment by TimoGlastra** [comment]
Can you open separate tickets for each item? That would make it easier to keep track of what is fixed, assignments, etc.. It would also allow us to have discussion on the specific features and close issue that we may not want to implement for AFJ rest

**Comment by TimoGlastra** [comment]
Fixed by #148
